### PR TITLE
[CIS-1157] Rename ChannelListController linking delegate funcs

### PR DESCRIPTION
**Pull request by b-onc**
And disable `NewChannelQueryUpdater` since delegate funcs take care of this now

### CODE REVIEW

**Review by evsaev — APPROVED** [review]
LGTM

File: Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift [hunk 1]
```diff
@@ -356,33 +357,27 @@ public protocol ChatChannelListControllerDelegate: DataControllerStateDelegate {
...
-    
-    /// The controller asks the delegate if the newly updated `ChatChannel` should be unlinked from this Controller's query.
-    /// Defaults to `false`
-    /// - Parameters:
-    ///   - controller: The controller,
-    ///   - shouldUnlinkUpdatedChannel: The newly updated `ChatChannel` instance. This instance is linked to the controller's query.
-    func controller(
-        _ controller: ChatChannelListController,
-        shouldUnlinkUpdatedChannel channel: ChatChannel
+        shouldListUpdatedChannel channel: ChatChannel
```
Comment (nuno-vieira) [inline Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift:380]: If these are breaking changes, shouldn't we deprecate them first? We are already in production 4.0.

Comment (b-onc) [reply Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift:380]: This wasn't released, only on main

Comment (nuno-vieira) [reply Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift:380]: Hum okay 🤔 Then if we fix the Channel List tests, it should be LGTM

**Review by tbarbugli — APPROVED** [review]

**Pull request merged**